resources/RemoveInactiveUsersRunner.php

```diff
@@ -57,36 +72,58 @@
     if ($lastLoginDate) { // null lastLoginDate check
         $interval = $today->diff($lastLoginDate);
     } else { // This might only be run once, since new users always have field filled.
-        echo "User has no lastLoginDate (it may have been a very long time.)\n";
-        echo "Deleting user.\n";
+        echo "Deleting this user as it has no last login date " .
+        "(it may have been a very long time). \n";
+        //writing to file to say who would have been deleted
+        file_put_contents("InactiveUsersToBeDeleted.txt", "The user " .
+        "with Id ".$user->getId()." would have been deleted as it " .
+        "has no last login date.\n", FILE_APPEND);
         deleteUser($user, $entityManager);
         echo "\n";
         // Move onto the next users.
         continue;
     }
 
-    $elapsedMonths = (int) $interval->format('%a') / 30;
-    echo 'Months elapsed since last login: ' . $elapsedMonths . "\n";
-
-    if ($elapsedMonths > $deletionThreshold) { // Delete user
-        echo "Deleting user\n";
-        deleteUser($user, $entityManager);
+    $elapsedMonths = (int) $interval->format('%a') / 30.4375; // 30.4375 is 365.25/12
+    echo 'Months elapsed since last login: ' . round($elapsedMonths,2) . "\n";
+    // Rounded months since last login to 2 decimal places for easier reading.
+
+    if ($elapsedMonths > $deletionThreshold) // Delete user
+        // Check if it is a dry run
+        if ($dryRun == '1') {
+            // Dry run option is set, so append user to be deleted to the file
+            echo "This user will be deleted when it isn't a dry run. \n\n";
+            file_put_contents("InactiveUsersToBeDeleted.txt", "The user with " .
```

Comment:
For the file output, it would be good if it was more machine readbale - so we can pipe it into other tools or open it in excel.

Maybe something like CSV

```
User, lastLoginDate, elapsedMonths
1, '10/11/2022 17:16:00', 11.00
2, '10/10/2022 17:16:00', 0.00
...
```

It would make the output easier to sort and easier to confirm that it's behaving sensibly.

Reply:
same for case where `lastLoginDate` is `null`

```
3, `null`, '?'
```

not really sure what's best to log for `elapsedMonths` when `lastLoginDate` is `null` - '?' doesn't feel to bad...

Reply:
ok